### PR TITLE
*: fix deadlock in new processor

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- When the new processor is enabled, deadlock can occur when the Unified Sorter is blocked due to the sink not being able to handle the large amount of data. 

### What is changed and how it works?
- Redesigned some details in the Unified Sorter so that the outputCh never blocks `AddEntry`.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### Release note

- Fix deadlock

### CODE REVIEW

File: cdc/puller/sorter/merger.go [hunk 1]
```diff
@@ -477,95 +508,7 @@ func printError(err error) error {
 		!strings.Contains(err.Error(), "context deadline exceeded") &&
 		cerrors.ErrAsyncIOCancelled.NotEqual(errors.Cause(err)) {
 
-		log.Warn("Unified Sorter: Error detected", zap.Error(err))
+		log.Warn("Unified Sorter: Error detected", zap.Error(err), zap.Stack("stack"))
```
Comment (amyangfei) [inline cdc/puller/sorter/merger.go:511]: What's the difference between the error stack and calling `zap.Stack` directly

File: cdc/puller/sorter/merger.go [hunk 2]
```diff
@@ -441,10 +443,39 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
...
+	errg.Go(func() error {
+		resolvedTsReceiver, err := resolvedTsNotifier.NewReceiver(time.Second * 1)
+		defer resolvedTsReceiver.Stop()
+
+		if err != nil {
+			if cerrors.ErrOperateOnClosedNotifier.Equal(err) {
+				log.Panic("unexpected error", zap.Error(err))
+			}
+			return errors.Trace(err)
+		}
```
Comment (amyangfei) [inline cdc/puller/sorter/merger.go:461]: check the error first, before the defer function. Besides why `ErrOperateOnClosedNotifier` is an unexpected error?

File: cdc/puller/sorter/merger_test.go [hunk 3]
```diff
@@ -480,115 +478,77 @@ func (s *sorterSuite) TestMergerCloseChannel(c *check.C) {
...
-// TestTaskBufferBasic tests the basic functionality of TaskBuffer
-func (s *sorterSuite) TestTaskBufferBasic(c *check.C) {
+// TestMergerOutputBlocked simulates a situation where the output channel is blocked for
+// a significant period of time.
+func (s *sorterSuite) TestMergerOutputBlocked(c *check.C) {
 	defer testleak.AfterTest(c)()
+	err := failpoint.Enable("github.com/pingcap/ticdc/cdc/puller/sorter/sorterDebug", "return(true)")
+	if err != nil {
+		log.Panic("Could not enable failpoint", zap.Error(err))
+	}
```
Comment (amyangfei) [inline cdc/puller/sorter/merger_test.go:488]: `c.Assert(err, check.IsNil)`

File: cdc/puller/sorter/merger.go (hunk 1 above)
Comment (liuzix) [reply cdc/puller/sorter/merger.go:511]: This error might not originate from this call stack.

**Review by amyangfei — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**